### DISCUSSION TITLE
classes/class_undoredo

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#20] [User-contributed notes]
# classes/class_undoredo

Inherits: Object Provides a high-level interface for implementing undo and redo operations. Description: UndoRedo works by registering methods and property changes inside "actions". You can create ...

https://docs.godotengine.org/en/latest/classes/class_undoredo.html

**Comment by kaprosdev** [comment]
Only actions within 800ms of each other are merged when using `MERGE_ENDS` or `MERGE_ALL` - which seems to be a hardcoded debounce period. 

(This was super helpful in my use case of "spinners where I want to consolidate rapid changes without clumping separate changes together" - but I imagine it could be confusing if you're trying to consolidate actions that won't occur in quick succession.)

**Comment by DustinReagan** [comment]
How can i get a list of all Callables currently in the current scene's History (in a godot editor script)?
